src/odemis/driver/nenovision.py

```
@@ -0,0 +1,479 @@
+
+    @staticmethod
+    def _check_nidaqmx() -> None:
+        """
+        Runs a safe canary check via a subprocess to ensure the NI-DAQmx C-library
+        will not segfault the main Odemis Python process if the driver is misconfigured.
+
+        :raises:
+            model.HwError: If the NI-DAQmx subprocess returns a fatal exit code.
+        """
+        canary_cmd = [sys.executable, "-c", "import nidaqmx; all(nidaqmx.system.System.local().devices)"]
+        proc = subprocess.run(canary_cmd)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)